cli/admin/src/main/java/io/pravega/cli/admin/readerGroup/ParseReaderGroupStreamCommand.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    @Override
+    public void execute() throws Exception {
+        ensureArgCount(4);
+        final String scope = getArg(0);
+        final String readerGroup = getArg(1);
+        final String segmentStoreHost = getArg(2);
+        final String fileName = getArg(3);
+        String stream = NameUtils.getStreamForReaderGroup(readerGroup);
+
+        @Cleanup
+        CuratorFramework zkClient = createZKClient();
+        @Cleanup
+        SegmentHelper segmentHelper = instantiateSegmentHelper(zkClient);
+        String tmpfilename = "tmp/RG_" + readerGroup + "_" + new Random().nextInt(1000);
+        readRGSegmentToFile(segmentHelper, segmentStoreHost, scope, stream, tmpfilename, fileName);
+        output("The readerGroup stream has been successfully written into %s", fileName);
+    }
+
+    /**
+     * Reads the contents of the segment starting from the given offset and writes into the provided file.
+     *
+     * @param segmentHelper       A {@link SegmentHelper} instance to read the segment.
+     * @param segmentStoreHost    Address of the segment-store to read from.
+     * @param scope               The name of the scope.
+     * @param stream              The name of the stream.
+     * @param fileName            A name of the file to which the data will be written.
+     * @throws IOException if the file create/write fails.
+     * @throws Exception if the request fails.
+     */
+    private void readRGSegmentToFile(SegmentHelper segmentHelper, String segmentStoreHost, String scope, String stream,
+                                     String tmpfilename, String fileName) throws IOException, Exception {
+
+        File outputfile = FileHelper.createFileAndDirectory(fileName);
+
+        @Cleanup
+        BufferedWriter writer = new BufferedWriter(new FileWriter(outputfile));
+
+        Controller controller = instantiateController();
+        CompletableFuture<StreamSegments> streamSegments = controller.getCurrentSegments(scope, stream);
+        StreamSegments ss = streamSegments.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        int segmentid = (ss.getNumberOfSegments() > 0) ? (ss.getNumberOfSegments() - 1) : 0;
+        String fullyQualifiedSegmentName = NameUtils.getQualifiedStreamSegmentName(scope, stream, segmentid);
+
+        CompletableFuture<WireCommands.StreamSegmentInfo> segmentInfo = segmentHelper.getSegmentInfo(fullyQualifiedSegmentName,
+                            new PravegaNodeUri(segmentStoreHost, getServiceConfig().getAdminGatewayPort()), authHelper.retrieveMasterToken(), 0L);
+        WireCommands.StreamSegmentInfo streamSegmentInfo = segmentInfo.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        long startOffset = streamSegmentInfo.getStartOffset();
+        long length = streamSegmentInfo.getWriteOffset();
+
+        // create a temp file and write contents of the segment into it
+        AdminCommandState state = new AdminCommandState();
```

Comment:
Why are you creating a new state? Is it wrong to reuse the one in this command itself?

Reply:
corrected